### ISSUE TITLE
Different implementation in Colored MNIST

### ISSUE THREAD

**Issue by lianqing11** [facebookresearch/InvariantRiskMinimization#4]
Hi author, 

Thanks for your wonderful work. I am confused about the implementation of the squared gradient norm. In the paper, you use X^{e,i} and X^{e,j} to compute the squared gradient norm, where X^{e,i} and X^{e,j} are two random minibatches. However, it seems that the squared gradient norm is computed with the use of X^{e,i}, and X^{e,i} in colored MNIST. Do these two kinds of implementation work in the same way? 

Best regards,
Qing LIAN

**Comment by igul222** [comment]
Thanks for your question! The different-minibatches technique is used to get an unbiased estimate of the loss for SGD. In this code, we use full-batch gradient descent rather than SGD, so it's not applicable.

**Comment by kirk86** [comment]
Hi @igul222 , 
sorry to bother you I just had a similar question


> we use full-batch gradient descent rather than SGD

and I was wondering why is that the case, isn't full batch more computationally intensive?

Also I'm not quite sure what would be the appropriate modifications in order for IRM to work on mini-batches with SGD? 
If we were to implement SGD using the penalty function on each mini-batch then autogad would throw an error.

Any ideas?

Thanks!
